src/plugins/dashboard/public/dashboard_api/get_dashboard_api.ts

```diff
@@ -171,6 +171,14 @@ export function getDashboardApi({
       unifiedSearchManager.internalApi.controlGroupReload$,
       unifiedSearchManager.internalApi.panelsReload$
     ).pipe(debounceTime(0)),
+    getSerializedState: async () => {
```

Comment:
https://github.com/elastic/kibana/pull/203662 made getState sync so this method could now become sync as well.

Reply:
Oh good call! Updated in [b58c33d](https://github.com/elastic/kibana/pull/204140/commits/b58c33dbc559eee7ff5b503e103c4951263214f2).

Unfortunately, the `getSerializedState` method still needs to be async for now since the `getDashboardState` method on the CM service [relies on a Promise to extract searchSource references](https://github.com/elastic/kibana/pull/204140/files#diff-11c75ee4e74ada316572f3b897515c774724e9b115035c40c895cc563af2070dR83-R97). But this might change when we move reference handling to the server in https://github.com/elastic/kibana/issues/192758.